### PR TITLE
Reckless: add rust installer and json output

### PR DESCRIPTION
**Pull request by endothermicdev**
This PR does several things:
1. Adds an installer for rust plugins.
2. Adds a --version flag.
3. Refactors logging and output.
4. Allows option flags in any order. (previously limited to before subcommands only)
5. Adds a --json output option.

In general, this prepares for a reckless-rpc plugin coming next.

### CODE REVIEW

**Review by ShahanaFarooqui — COMMENTED** [review]
- Search without result does not show any message, for example:  `reckless search nonexistingplugin` should show a message stating that `plugin not found`/`plugin nonexistingplugin not found`

- holdinvoice install command throws below error (`reckless --regtest --verbose --json install holdinvoice`)
```
Traceback (most recent call last):
  File "/usr/local/bin/reckless", line 1803, in <module>
    log.add_result(args.func(target))
  File "/usr/local/bin/reckless", line 1314, in install
    installed = _install_plugin(src)
  File "/usr/local/bin/reckless", line 1245, in _install_plugin
    staged_src = INSTALLER.dependency_call(staged_src)
  File "/usr/local/bin/reckless", line 936, in cargo_installation
    log.error(cargo.stderr.read())
AttributeError: 'str' object has no attribute 'read'
```

- Can `reckless install` accept plugin options too, like `summars-columns`, `summars-sort-by` with `reckless install summars`?

File: tools/reckless [hunk 1]
```diff
@@ -827,34 +908,66 @@ def install_to_python_virtual_environment(cloned_plugin: InstInfo):
...
+                    stderr=PIPE, text=True)
+
+    if cargo.returncode == 0:
+        log.debug('rust project compiled successfully')
+    else:
+        if cargo.stderr:
+            log.error(cargo.stderr.read())
+        else:
+            log.error('error encountered during build, cargo exited'
+                      f'with return code {cargo.returncode}')
```
Comment (ShahanaFarooqui) [inline tools/reckless:943]: nitpick: Can use shorthand `log.error(cargo.stderr.read() if cargo.stderr else f'error encountered during build, cargo exited with return code {cargo.returncode}')`

File: tools/reckless [hunk 2]
```diff
@@ -827,34 +908,66 @@ def install_to_python_virtual_environment(cloned_plugin: InstInfo):
...
-python3venv = Installer('python3venv', 'text/x-python', exe='python3',
+def cargo_installation(cloned_plugin: InstInfo):
+    call = ['cargo', 'build', '--release', '-vv']
+    # FIXME: the symlinked Cargo.toml allows the installer to identify a valid
+    # plugin directory, but is unneeded, and actually confuses cargo if not
+    # removed prior to installing.
+    (Path(cloned_plugin.source_loc) / 'Cargo.toml').unlink()
+
+    # source_loc now contains a symlink to the entrypoint and 'source/plugin/'
+    source = Path(cloned_plugin.source_loc)/'source'/cloned_plugin.name
```
Comment (ShahanaFarooqui) [inline tools/reckless:928]: nitpick: add spaces before & after "/": `source = Path(cloned_plugin.source_loc) / 'source' / cloned_plugin.name`

**Comment by endothermicdev** [comment]
> Search without result does not show any message
Good catch, I changed the default logging level and missed changing this one to `logging.info` fixed!

> holdinvoice install command throws below error
Ah, right, I changed the call from Popen to run, which has the outputs strings.  Also raised the right error to catch here.

**Comment by endothermicdev** [comment]
>Can reckless install accept plugin options too, like summars-columns, summars-sort-by with reckless install summars?

I like this idea.  We could pass these in a list to an optional keyword, and then maybe track the options in the metadata.  Is there a good way to validate the plugins options though?  Any errors will cause the node to not start.

**Review by rustyrussell — APPROVED** [review]
Ack 87895bd99f25bcc72d2c3db58fd7829a166290ad

**Pull request merged**